### PR TITLE
Fix xctestrun file not being found when no arch is present in the filename

### LINKED ISSUE #1105: Error on patrol test " No .xctestrun file was found"
**Describe the bug**

When I run command:
`patrol test --flavor qa` I get "No .xctestrun file was found at .../build/ios_integ/Build/Products

My tests were successfully running few weeks ago, with different versions and patrol drive.

**To Reproduce**

1. Run `patrol test --flavor qa`
2. Logs: 

No device specified, using the first one (iPhone 14 Pro)
Every test target will be run 1 time(s)
• Building app with entrypoint authentication_test.dart for iOS simulator...
✓ Completed building app with entrypoint authentication_test.dart for iOS simulator (37.7s)
• Running app with entrypoint authentication_test.dart for iOS simulator on simulator iPhone 14 Pro...
Error: No .xctestrun file was found at .../build/ios_integ/Build/Products
See the logs above to learn what happened. Also consider running with --verbose. If the logs still aren't useful, then it's a bug - please report it.
 FAIL  authentication_test.dart on iPhone 14 Pro

**Expected behavior**

The test should run.

**Device information**

- Device: iPhone 14 pro simulator
- OS version: iOs 16.2
- Does it also both on physical and virtual devices? [Yes, no, didn't check] Didn't check

<details>
<summary>Additional information</summary>

Flutter 3.7.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 12cb4eb7a0 (3 weeks ago) • 2023-03-01 10:29:26 -0800
Engine • revision ada363ee93
Tools • Dart 2.19.3 • DevTools 2.20.1

patrol: ^1.0.8

patrol_cli v1.1.0

</details>

### PR DESCRIPTION
**Pull request by bartekpacia**
Fixes #1105

### CODE REVIEW

File: packages/patrol_cli/test/ios/ios_test_backend_test.dart [hunk 1]
```diff
@@ -1,40 +1,177 @@
...
+        String? arch,
+      }) {
+        test(description, () async {
+          final targetPlatform = simulator ? 'iphonesimulator' : 'iphoneos';
+          var targetArch = arch;
+          if (targetArch != null) {
+            targetArch = '-$targetArch';
+          }
+
+          final name = '${scheme}_${targetPlatform}16.2$targetArch.xctestrun';
```
Comment (bartekpacia) [inline packages/patrol_cli/test/ios/ios_test_backend_test.dart:80]: I don't like having this in test, it's essentially duplicated code from `IOSTestBackend.xcTestRunPath`.

File: packages/patrol_cli/test/ios/ios_test_backend_test.dart
Comment (bartekpacia) [inline packages/patrol_cli/test/ios/ios_test_backend_test.dart:1]: I don't like the result. I think the best solution would be to specify the matrix of cases, and then dynamically create test cases? Is that even possible in Dart?


```
simulator: true, false
scheme: Runner, dev
arch: none, arm64, x86_64, arm64_x86_64
```

Comment (Albert221) [reply packages/patrol_cli/test/ios/ios_test_backend_test.dart:1]: Yeah, it's called `for` loop :D

File: packages/patrol_cli/test/ios/ios_test_backend_test.dart [hunk 2]
```diff
@@ -1,40 +1,177 @@
...
+        fs: fs,
+        iosDeploy: FakeIOSDeploy(),
+        parentDisposeScope: DisposeScope(),
+        logger: FakeLogger(),
+      );
+    });
+
+    group('xcTestRunPath', () {
+      @isTest
+      void testXcTestRunPath(
```
[the remark this answers is not in the document]
Comment (Albert221) [reply packages/patrol_cli/test/ios/ios_test_backend_test.dart:67]: I think this is fine

**Pull request merged**